SummarizedExperiment support

vsuksi opened the pull request:
Implemented SummarizedExperiment-support, see commits for more context about changes. In brief, it could have made sense to completely refactor the package and create methods separately for MetaboSet and SummarizedExperiment. However, as MetaboSet is deprecated, it is likely preferrable to handle only SummarizedExperiment objects, which would make separate methods not worth it for simplicity. The translation is quite minimal: multiple assays is supported, but this mode of operation is not showcased in examples/vignettes as now it holds for both SummarizedExperiment and MetaboSet. Definitely need fresh eyes on this, thanks!

attlih commented on R/batch_correction.R:
```diff
@@ -186,71 +204,93 @@ perform_repeatability <- function(object, group) {
-#' batch_aligned <- align_batches(example_set_na, example_set_fill, 
+#' assay(ex_set_na)[2, c(1, 7, 13, 19)] <- NA
+#' assay(ex_set_na)[3, c(26, 32, 38, 44)] <- NA
+#' batch_aligned <- align_batches(ex_set_na, ex_set_fill, 
```
Example throws an error:

```
Error in batchCorr::alignBatches(peakInfo = p_info, PeakTabNoFill = t(assay(object_na)),  : 
  unused arguments (NAhard = NAhard, reportPath = plot_folder)
```

attlih reviewed (changes requested):
Not a complete review, just a quick overlook

antagomir commented on R/stats.R:
```diff
@@ -1301,157 +1301,173 @@ perform_oneway_anova <- function(object, formula_char,
   }
-  rownames(results_df) <- results_df$Feature_ID
+  # Keep features without errors
+  results_df <- data.frame(Feature_ID = features, 
```
@vsuksi also make sure sufficient unit tests on this and other things are included in tests/

antagomir commented on R/utils.R:
```diff
@@ -361,3 +365,39 @@ prop_found <- function(x) {
+    stop("When using multiple assays, specify assay.type", call. = FALSE)
+  } else if (!assay.type %in% names(assays(object)) & assay.type != 1) {
+    stop(assay.type, " was specified but not found in assays", call. = FALSE)
+  }
+  assay.type
```
check here and elsewhere. Can be a separate issue after the first PR

antagomir reviewed (commented):
Overall looks good. 

Some suggestions for new issues after the PR.

Unit tests: make sure there is some testing that results from old and new versions match.

Rest of the modifications are better done after the PR when the package can be tested further. Among these are questions about better syncing the pkg with ggplot, dplyr/tidyOmics, other R/Bioc metabolome, and mia frameworks

attlih commented on LICENSE:
```diff
@@ -1,2 +1,21 @@
-COPYRIGHT HOLDER: Anton Klåvus, Jussi Paananen, Oskari Timonen at University of Eastern Finland & Afekta Technologies
+MIT License
+
+Copyright (c) 2020 Anton Klåvus, Jussi Paananen, Oskari Timonen at University of Eastern Finland & Afekta Technologies
```
Remember to ask permission from original authors if this is changed.

vsuksi commented:
I would go ahead and merge this and keep working on tangential issues in separate branches if this is ok.

antagomir commented:
Ok to me but I think it would be good to have OK also from @RetuWonData or someone else from that end. 

Seems essential to me to merge this so that it is possible to make progress and start to develop more well defined parts of this.

antagomir commented:
Can you "resolve" those earlier feedbacks from us that have been dealt with?

RetuWonData reviewed (commented):
Sorry to keep you waiting. So much work here, great job! When scrolling through everything seems good. I spotted two minor things while playing around with it.

Merging is okay for me! Would it make sense to include a link to SE info?

vsuksi commented on NAMESPACE:
```diff
@@ -81,7 +86,6 @@ export(read_from_excel)
 export(ruvs_qc)
 export(save_batch_plots)
 export(save_beeswarm_plots)
-export(save_dc_plots)
```
(This change was accidentally included in this branch after omitting it earlier, I could have rebased this branch but that would break your comments, methinks)

The same functionality is available in the correct_drift wrapper. The idea here was to de-export low-level functionality for drift correction (dc_cubic_spline, inspect_dc, save_dc_plots, not used in original tutorials/template) to not confuse the user and provide a more consistent interface. Same for cluster_features. Shouldn't be a problem to revert this, though.

RetuWonData replied:
I see! Maybe this de-exporting is a good way to go.

vsuksi commented:
> Can you "resolve" those earlier feedbacks from us that have been dealt with? 

@antagomir I resolved the addressed issues, the rest are best done in separate branches. I included more notes on feedback in the commit message for "Address immediately applicable feedback", for some reason.

attlih commented on R/stats.R:
```diff
@@ -620,16 +641,16 @@ perform_auc <- function(object, time = time_col(object),
-                                     subject_col = subject) %>%
-    merge_metabosets()
+  # Construct new SummarizedExperiment object (with all modes together)
+  new_object <- SummarizedExperiment(assays = aucs, 
```
Is it intended that this always returns SummarizedExperiment?

vsuksi replied:
Oops, fixed!

attlih reviewed (commented):
Overall looks good! Although I still get quite many warnings and errors running tests/examples, so I assume those are not ready yet.

vsuksi commented:
Feel free to provide details on the errors and warnings @attlih . Using R development version (4.5) and Bioc-devel on Linux and Windows and submission-specific [flags](https://contributions.bioconductor.org/general.html?q=flags#r-cmd-check-environment), I get no errors or warnings in CMD check or BiocCheck (after removing supraHex and batchCorr functionality). There are several notes but they are sort of trivial things I'm addressing in specific branches to be merged for the submission. 

I was trying to streamline the checking using rworkflows so such matters would be easier, but it doesn't seem trivial to set up for use with the flags used in submission building and checking. Help on this would be appreciated, or we simply proceed as per my setup on Linux and Windows and resolve remaining problems in the actual submission building and checking.

attlih commented:
To clarify, I ran `devtools::test()` and `devtools::run_examples()`. There's only one error which is an easy fix. This probably fails as `grDevices::pdf` is not able to create document to a folder which does not exist. 

```
Error (test_plotting.R:114:3): Batch plots work with and without multiple assays
Error in `grDevices::pdf(file, width = width, height = height)`: cannot open file 'C:\Users\ATTELI~1\AppData\Local\Temp\RtmpITEg4c\test\batch_plots.pdf'
Backtrace:
    ▆
 1. └─notame::save_batch_plots(...) at test_plotting.R:114:3
 2.   └─grDevices::pdf(file, width = width, height = height) at notame-bioc/R/batch_correction.R:477:3
```

Warnings I get are not related to fonts but more to test design, although I agree that these are not critical. For example this:

```
Warning (test_stats.R:673:3): Simple tests return all features despite errors for single features
coercing argument of type 'list' to logical
Backtrace:
    ▆
 1. └─testthat::expect_true(all(lapply(res[1, -1], function(col_value) is.na(col_value)))) at test_stats.R:673:3
 2.   └─testthat::quasi_label(enquo(object), label, arg = "object")
 3.     └─rlang::eval_bare(expr, quo_get_env(quo))
```

Automated unit tests would be beneficial indeed but currently I don't have any time to help with that.

vsuksi commented:
Alright, thanks. The pdf error seems setup-specific, so I'll go ahead and merge this branch (we have the startup note for a reason). Trying to get rworkflows to work with the submission flags for R CMD check can be such a timesink for me that I'm planning to do the checks as per my setup and then address final setup-dependent issues in the actual submission.
